eng/native/configurecompiler.cmake

```diff
@@ -243,6 +245,8 @@ if (CLR_CMAKE_HOST_WIN32)
   set(STATIC_MT_CPP_LIB  "libcpmt$<$<OR:$<CONFIG:Debug>,$<CONFIG:Checked>>:d>.lib")
 endif(CLR_CMAKE_HOST_WIN32)
 
+add_definitions(-D_FILE_OFFSET_BITS=64)
```

Comment:
Are there any interop places that assume offsets are 32-bit long? I've definitely seen things that takes 32-bit integers for memory buffer sizes (where a size_t would be preferable), so I'm worried that something like this might become an issue.

Reply:
I was wondering the same that how it is being unconditionally set in coreclr while in libraries, it is only set to 64 for 32-bit platforms. I am trying to share some chunks from coreclr with installer and libraries via eng/native, but this block has been reverted (because libraries and installer couldn't make use of it due to how conditions are structured cmake's CXX vs. C variables).

Your concern is, however, still valid for `src/coreclr/CMakeLists.txt` in master branch, where this is set to 64 for all platforms.

Not sure if it impacts arm (32-bit) the same way as i386, but currently there is no i386 CI leg, so it might be some misalignment in CoreCLR due to that.